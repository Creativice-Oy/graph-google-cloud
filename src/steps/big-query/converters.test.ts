import {
  createBigQueryDatasetEntity,
  createBigQueryTableEntity,
} from './converters';
import {
  getMockBigQueryDataset,
  getMockBigQueryTable,
} from '../../../test/mocks';
import { DEFAULT_INTEGRATION_CONFIG_PROJECT_ID } from '../../../test/config';

describe('#createBigQueryDatasetEntity', () => {
  test('should convert to entity', () => {
    expect(
      createBigQueryDatasetEntity(getMockBigQueryDataset()),
    ).toMatchSnapshot();
  });

  test('should convert to entity with public param set to on', () => {
    expect(
      createBigQueryDatasetEntity(
        getMockBigQueryDataset({
          access: [
            { role: 'WRITER', specialGroup: 'allAuthenticatedUsers' },
            { role: 'WRITER', specialGroup: 'projectWriters' },
            { role: 'OWNER', specialGroup: 'projectOwners' },
            { role: 'OWNER', userByEmail: 'admin@creativice.com' },
            { role: 'READER', iamMember: 'allUsers' },
            { role: 'READER', specialGroup: 'projectReaders' },
          ],
        }),
      ),
    ).toMatchSnapshot();
  });
});

describe('#createBigQueryTableEntity', () => {
  test('should convert to entity', () => {
    expect(
      createBigQueryTableEntity({
        data: getMockBigQueryTable(),
        projectId: DEFAULT_INTEGRATION_CONFIG_PROJECT_ID,
        isPublic: false,
      }),
    ).toMatchSnapshot();
  });
});
