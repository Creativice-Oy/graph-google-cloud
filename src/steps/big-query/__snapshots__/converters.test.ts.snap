// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createBigQueryDatasetEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "DataStore",
    "Database",
  ],
  "_key": "j1-gc-integration-dev-v2:test_big_query_dataset",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "access": Array [
          Object {
            "role": "WRITER",
            "specialGroup": "projectWriters",
          },
          Object {
            "role": "OWNER",
            "specialGroup": "projectOwners",
          },
          Object {
            "role": "OWNER",
            "userByEmail": "admin@creativice.com",
          },
          Object {
            "role": "READER",
            "specialGroup": "projectReaders",
          },
        ],
        "creationTime": "1610919167542",
        "datasetReference": Object {
          "datasetId": "test_big_query_dataset",
          "projectId": "j1-gc-integration-dev-v2",
        },
        "etag": "Q1RJczpo1aoPRRoUJ1pElQ==",
        "id": "j1-gc-integration-dev-v2:test_big_query_dataset",
        "kind": "bigquery#dataset",
        "lastModifiedTime": "1610919344642",
        "location": "US",
        "selfLink": "https://www.googleapis.com/bigquery/v2/projects/j1-gc-integration-dev-v2/datasets/test_big_query_dataset",
      },
    },
  ],
  "_type": "google_bigquery_dataset",
  "classification": null,
  "createdOn": 1610919167542,
  "description": undefined,
  "displayName": "test_big_query_dataset",
  "encrypted": true,
  "etag": "Q1RJczpo1aoPRRoUJ1pElQ==",
  "id": "j1-gc-integration-dev-v2:test_big_query_dataset",
  "kmsKeyName": undefined,
  "location": "US",
  "name": "test_big_query_dataset",
  "public": false,
  "updatedOn": 1610919344642,
  "webLink": "https://console.cloud.google.com/bigquery?project=j1-gc-integration-dev-v2&d=test_big_query_dataset&p=j1-gc-integration-dev-v2&page=dataset",
}
`;

exports[`#createBigQueryDatasetEntity should convert to entity with public param set to on 1`] = `
Object {
  "_class": Array [
    "DataStore",
    "Database",
  ],
  "_key": "j1-gc-integration-dev-v2:test_big_query_dataset",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "access": Array [
          Object {
            "role": "WRITER",
            "specialGroup": "allAuthenticatedUsers",
          },
          Object {
            "role": "WRITER",
            "specialGroup": "projectWriters",
          },
          Object {
            "role": "OWNER",
            "specialGroup": "projectOwners",
          },
          Object {
            "role": "OWNER",
            "userByEmail": "admin@creativice.com",
          },
          Object {
            "iamMember": "allUsers",
            "role": "READER",
          },
          Object {
            "role": "READER",
            "specialGroup": "projectReaders",
          },
        ],
        "creationTime": "1610919167542",
        "datasetReference": Object {
          "datasetId": "test_big_query_dataset",
          "projectId": "j1-gc-integration-dev-v2",
        },
        "etag": "Q1RJczpo1aoPRRoUJ1pElQ==",
        "id": "j1-gc-integration-dev-v2:test_big_query_dataset",
        "kind": "bigquery#dataset",
        "lastModifiedTime": "1610919344642",
        "location": "US",
        "selfLink": "https://www.googleapis.com/bigquery/v2/projects/j1-gc-integration-dev-v2/datasets/test_big_query_dataset",
      },
    },
  ],
  "_type": "google_bigquery_dataset",
  "classification": null,
  "createdOn": 1610919167542,
  "description": undefined,
  "displayName": "test_big_query_dataset",
  "encrypted": true,
  "etag": "Q1RJczpo1aoPRRoUJ1pElQ==",
  "id": "j1-gc-integration-dev-v2:test_big_query_dataset",
  "kmsKeyName": undefined,
  "location": "US",
  "name": "test_big_query_dataset",
  "public": true,
  "updatedOn": 1610919344642,
  "webLink": "https://console.cloud.google.com/bigquery?project=j1-gc-integration-dev-v2&d=test_big_query_dataset&p=j1-gc-integration-dev-v2&page=dataset",
}
`;

exports[`#createBigQueryTableEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "DataObject",
  ],
  "_key": "j1-gc-integration-dev-v2:new_dataset_test.mascots",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "creationTime": "1621610438123",
        "id": "j1-gc-integration-dev-v2:new_dataset_test.mascots",
        "kind": "bigquery#table",
        "tableReference": Object {
          "datasetId": "new_dataset_test",
          "projectId": "j1-gc-integration-dev-v2",
          "tableId": "mascots",
        },
        "type": "TABLE",
      },
    },
  ],
  "_type": "google_bigquery_table",
  "classification": null,
  "createdOn": 1621610438123,
  "displayName": "mascots",
  "expirationTime": undefined,
  "friendlyName": undefined,
  "id": "j1-gc-integration-dev-v2:new_dataset_test.mascots",
  "name": "mascots",
  "public": false,
  "type": "TABLE",
  "webLink": "https://console.cloud.google.com/bigquery?project=j1-gc-integration-dev-v2&d=new_dataset_test&p=j1-gc-integration-dev-v2&t=mascots&page=table",
}
`;
